karaf/s2i/run

```diff
@@ -3,6 +3,19 @@
 # Global S2I variable setup
 source `dirname "$0"`/s2i-setup
 
+# Setup Jolokia to accept basic auth, using a randomly generated password that is stored
+# in the container in the ${DEPLOYMENTS_DIR}/jolokia.pw file.
+AB_JOLOKIA_USER="jolokia"
+if [ -f "${DEPLOYMENTS_DIR}/jolokia.pw" ] ; then
+	AB_JOLOKIA_PASSWORD=`cat "${DEPLOYMENTS_DIR}/jolokia.pw"`
+else
+	AB_JOLOKIA_PASSWORD=`tr -cd '[:alnum:]' < /dev/urandom | fold -w30 | head -n1`
+	touch "${DEPLOYMENTS_DIR}/jolokia.pw"
+	chmod 600 "${DEPLOYMENTS_DIR}/jolokia.pw"
+	echo $AB_JOLOKIA_PASSWORD > "${DEPLOYMENTS_DIR}/jolokia.pw"
+fi
+export AB_JOLOKIA_USER
```

Comment:
Your right, we need to update [it](https://github.com/rhuss/jolokia/blob/7a8249cc1e36680aee26a8ddd570520eef3e190b/agent/jvm/src/main/java/org/jolokia/jvmagent/JolokiaServer.java#L415-L415) so that it uses setWantClientAuth() instead of setNeedClientAuth().  I'll send in a follow up PR for that.